### ISSUE TITLE
Multiple Applications(Spark Contexts) Concurrently Fail With Broadcast Error

### ISSUE THREAD

**Issue by ryaminal** [spark-jobserver/spark-jobserver#18]
We are unable to run more than one application at a time using Spark 1.0.0 on CDH5. We submit two applications using two different SparkContexts on the same Spark Master. The Spark Master was started using the following command and parameters and is running in standalone mode:
<quote>
<pre>
/usr/java/jdk1.7.0_55-cloudera/bin/java 
  -XX:MaxPermSize=128m 
  -Djava.net.preferIPv4Stack=true 
  -Dspark.akka.logLifecycleEvents=true 
  -Xms8589934592 
  -Xmx8589934592 
  org.apache.spark.deploy.master.Master 
    --ip ip-10-186-155-45.ec2.internal
</pre>
</quote>

When submitting this application by itself it finishes and all of the data comes out happy. The problem occurs when trying to run another application while an existing application is still processing and we get an error stating that the spark contexts were shut down prematurely.
</br>
</br>
The errors can be viewed in the following pastebins. All IP addresses have been changed to 1.1.1.1 for security reasons. 
Notice that on the top of the logs we have printed out the spark config stuff for reference.

The working logs: <a href="http://pastebin.com/CnitnMhy">Working Pastebin</a>
The broken logs: <a href="http://pastebin.com/VGs87bBZ">Broken Pastebin</a>

We have also included the worker logs. For the second app, we see in the work/app/ directory 7 additional directors: `0/ 1/ 2/ 3/ 4/ 5/ 6/`. There are then two different groups of errors. The first three are one group and the other 4 are the other group of errors.
<pre>
Worker log for broken app group 1: <a href="http://pastebin.com/7VwZ1Gwu">Broken App Group 1</a>
Worker log for broken app group 2: <a href="http://pastebin.com/shs4d8T4">Broken App Group 2</a>
Worker log for working app: available upon request.
</pre>

The two different errors are the last lines of both groups and are:
<quote><pre>Received LaunchTask command but executor was null</pre></quote><quote><pre>Slave registration failed: Duplicate executor ID: 4</pre></quote>

tl;dr
We are unable to run more than one application in the same spark master using different spark contexts. The only errors we see are broadcast errors.

**Comment by dannylinden** [comment]
i ran into the same issue with CDH5.3 and Spark on Yarn-Client. HAve you found a solution/workaround @ryaminal ?

**Comment by emmekappa** [comment]
Have the same issue here.

**Comment by velvia** [comment]
Going to shelve this issue into #47 and PR #120 ....
